### DISCUSSION TITLE
[V5] Camelcase not being recognized/prioritized

### DISCUSSION THREAD

**Discussion by yarapolana** [adonisjs/core#2428] [Help] [answered]
I was researching how to serialize values to be camelcase on AdonisJS, and I found an old article/post for v4 where someone suggested changing the values in the database instead of serializing every request. I did that for my user to test, changed `username` to `userName` on my migration, model, etc. 
When creating a new user it gives me an error, `user_name` does not exist although that's not what I want.
Is there any config that I am missing? Or should I just serialize everything I want in camelcase?

Error
```
error: insert into "users" ("created_at", "email", "password", "updated_at", "user_name") values ($1, $2, $3, $4, $5) returning "id" - column "user_name" of relation "users" does not exist
```

Controller
<img width="267" alt="Screenshot 2021-03-26 at 20 06 11" src="https://user-images.githubusercontent.com/19730118/112686998-b76f3f00-8e6e-11eb-862a-eaa14e3d1964.png">

Model
<img width="365" alt="Screenshot 2021-03-26 at 19 59 42" src="https://user-images.githubusercontent.com/19730118/112686412-cef9f800-8e6d-11eb-840d-ed7741606399.png">

Migration
<img width="657" alt="Screenshot 2021-03-26 at 20 00 05" src="https://user-images.githubusercontent.com/19730118/112686450-de794100-8e6d-11eb-8267-f3728ec6d5dd.png">

**Comment by RomainLanz** [comment]
Hey @yarapolana! 👋 

In AdonisJS 5, we are following the SQL convention.
You should have (by default) everything in snake_case inside your database.
In your JS file, we are automatically converting everything to camelCase.

**Reply by yarapolana** [reply]
@RomainLanz 👋🏽 Do you recommend serializing everything to camelcase in the Model? or in the Controller? I remember having transforms in V4, is that a thing in v5?

**Comment by thetutlage** [comment, marked as answer]
There are a couple of things here.

- **Serialization** is the process of converting a model ( a Javascript class ) to a JSON object. This is something you usually do when creating a REST API, since you need to send JSON over the wire and we talk about serialization here https://preview.adonisjs.com/guides/models/serializing-models
- **Column names**: A very popular convention is to use `snake_case` column names in the database and `camelCase` property names in Javascript. This is what we optimize for by default.

When you define a column `userName` on the model. We assume that it will be represented as `snake_case` in the database. Now this assumption can be wrong, based upon how you have structured your database.

If its just one column, then you can define its column name inside the database within the decorator. In the following example, both the SQL and the Javascript has the same column name.

```ts
@column({ columnName: 'userName' })
public userName: string
```

If your entire database is in `camelCase`, then you can overwrite the automatic conversion at the global level within the `config/database.ts` file.

```ts
  orm: {
    getColumnName(_model, key) {
      return key
    },
  },
```

**Reply by yarapolana** [reply]
Thank you so much for the reply.
Missed the columnName 🥲🥴, I was checking out orm but seems to be deprecated, I read in the releases briefly, about something called "naming... strategy?" dont remember at the moment.
Will that work the same as orm?

**Comment by janainascal** [comment]
HI there, I'd like to hijack this question: Is there a config for the same thing (keeping the camelCase) but for model serialization?
